bin/test.py

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python
```

Comment:
I could not run this script:

```console
~$ ./test.py
Testing stacks
AWS session created
Ensure key pair in region: ap-south-1
Ensure S3 app bucket in region: ap-south-1
Traceback (most recent call last):
  File "./test2.py", line 281, in <module>
    main()
  File "./test2.py", line 274, in main
    ensure_foundation(session, config)
  File "./test2.py", line 212, in ensure_foundation
    if not has_bucket(s3_client, app_bucket_name):
  File "./test2.py", line 73, in has_bucket
    raise ce
botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the GetBucketLocation operation: Access Denied
```

As such, I didn't spend too much time reading through it.

Reply:
Yeah, I am not currently not doing much in terms of credential validation, but I will change that in the future.

Reply:
Backlog SIM created

Reply:
Gotcha. FWIW, I suspect this is specifically due to my account not having access to get `GetBucketLocation` on the region-specific bucket resource.